### PR TITLE
Run sbt commands in CI via a script

### PR DESCRIPTION
**Pull request by jacobwinch**
## What does this change?

Addresses https://github.com/guardian/mobile-save-for-later/pull/64#issuecomment-1135721409. Moving all CI configuration into version control is generally desirable. For example, it will make it easier to review & apply all changes related to https://github.com/guardian/mobile-save-for-later/pull/64, as there will no longer be additional steps required in TeamCity.

I've [deployed this to `CODE`](https://riffraff.gutools.co.uk/deployment/view/fcd32028-95fa-4044-aa98-2b35af3d0b6c) and confirmed that it works as expected.

Once this PR is merged I will tidy up the TeamCity build configuration.

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "0.14.6")
 
-addSbtPlugin("com.gu" % "sbt-riffraff-artifact" % "1.1.7")
+addSbtPlugin("com.gu" % "sbt-riffraff-artifact" % "1.1.18")
```
Comment (jacobwinch) [inline project/plugins.sbt:3]: When using the old version TeamCity wasn't picking up the build number correctly; this upgrade fixes that problem.

File: script/ci [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+set -e
+
+sbt "project mobile-save-for-later; clean; compile; riffRaffUpload"
```
Comment (jacobwinch) [inline script/ci:5]: This is currently running [via this build step](https://teamcity.gutools.co.uk/admin/editRunType.html?id=buildType:Mobile_MobileSaveForLater&runnerId=RUNNER_135&cameFromUrl=%2Fadmin%2FeditBuildRunners.html%3Fid%3DbuildType%253AMobile_MobileSaveForLater%26init%3D1&cameFromTitle=), which I've disabled for now.

File: mobile-save-for-later/script/ci [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+set -e
+
+sbt "project mobile-save-for-later; clean; compile; riffRaffUpload"
```
Comment (jacobwinch) [inline mobile-save-for-later/script/ci:5]: This is currently running [via this build step](https://teamcity.gutools.co.uk/admin/editRunType.html?id=buildType:Mobile_MobileSaveForLater&runnerId=RUNNER_135&cameFromUrl=%2Fadmin%2FeditBuildRunners.html%3Fid%3DbuildType%253AMobile_MobileSaveForLater%26init%3D1&cameFromTitle=), which I've disabled for now.

**Review by DavidLawes — APPROVED** [review]
👍  makes sense to me

I notice that other projects we have (e.g. mapi, mobile-save-for-later-user-deletion) define their build steps in team city, as opposed to defining them in code. Is this a task we need on our backlog?

**Comment by jacobwinch** [comment]
> I notice that other projects we have (e.g. mapi, mobile-save-for-later-user-deletion) define their build steps in team city, as opposed to defining them in code. Is this a task we need on our backlog?

It would be good to do this but it's not especially urgent. Hopefully we will start moving Scala projects to GitHub Actions in the next few months (we aren't recommending this to all teams just yet due to a few performance concerns) and that migration will solve this type of problem too.

**Pull request merged**